packages/core/strapi/lib/commands/actions/ts/generate-types/command.js

```diff
@@ -10,12 +10,7 @@ module.exports = ({ command }) => {
   command
     .command('ts:generate-types')
     .description(`Generate TypeScript typings for your schemas`)
-    .option(
-      '-o, --out-dir <outDir>',
-      'Specify a relative directory in which the schemas definitions will be generated'
-    )
```

Comment:
Are we removing -o because we will now require the schema to be in a specific location? We may want to leave it in case someone is using it, but add a warning when it's used that types are required to be in a specific location

Reply:
Although, allowing this means that we'll have broken paths from other files

Reply:
I mean, keep this as an option for someone who wants to generate it in another location, but they'll know "this won't make my strapi work"

And in fact, I could see it being important to test things out without breaking their existing types. They generate it once in the correct location, then generate another one that uses their "new" types and can compare before deciding to overwrite

Reply:
yeah i think we need to get in good practice of deprecating over removing, removing annoys people, deprecating doesn't as much imo

Reply:
I'm honestly not sure for this one as we're reworking completely the feature and it doesn't work the same as before (at all).
But... If we keep it, we have to keep in mind that it will break everything when using it :man_shrugging:

Reply:
I would leave this one in, I can see several use-cases for it -- I've used it myself to test different versions of the generation and then run a diff against the files. But any time it's used, we should log a warning that Strapi only reads files from a specific location.

Reply:
Can you not rewrite the logic of the command to work with the new stuff you're doing? Then deprecate the command? I.e removing stuff is technically a breaking change.

Alternatively we could support both ways and live with the burden for now?

Reply:
[`489e422` (#16757)](https://github.com/strapi/strapi/pull/16757/commits/489e422f60fac25b346e270b8eeab5e4c3c19e46)